plugin/src/main/java/org/opensearch/ml/model/MLModelGroupManager.java

```diff
@@ -152,7 +185,39 @@ public void createModelGroup(MLRegisterModelGroupInput input, ActionListener<Str
                                                     indexResponse.getResult(),
                                                     indexResponse.getId()
                                                 );
-                                            wrappedListener.onResponse(r.id());
+
+                                            // TODO: Remove this feature flag check once feature is GA, as it will be enabled by default
+                                            if (isResourceSharingFeatureEnabled) {
+                                                // Create an entry in resource-sharing index
+                                                String modelGroupId = indexResponse.getId();
+                                                String modelGroupIndex = indexResponse.getIndex();
+                                                SharedWithActionGroup.ActionGroupRecipients recipients =
+                                                    new SharedWithActionGroup.ActionGroupRecipients(recipientMap.get());
+
+                                                ResourceSharingClient resourceSharingClient = ResourceSharingClientAccessor
+                                                    .getInstance()
+                                                    .getResourceSharingClient();
+
+                                                resourceSharingClient
+                                                    .share(
```

Comment:
Once onboarded and ml access framework is deprecated, what happens to the already existing resources with access controls defined by ml-plugin

Reply:
2 things:
1. This feature is currently for fresh 3.1 clusters only
2.We have a migration path (APIs) which will implement the migrate API to enable feature usage on existing clusters migrating to 3.1

Once the feature is enabled by default, ml-access framework can be safely deleted.